ext/fetch/27_eventsource.js

```diff
@@ -186,165 +192,162 @@ class EventSource extends EventTarget {
+    clearTimeout(this[_reconnectionTimerId]);
   }
 
   async [_loop]() {
-    let lastEventIDValue = "";
-    while (this[_readyState] !== CLOSED) {
-      const lastEventIDValueCopy = lastEventIDValue;
-      lastEventIDValue = "";
-      const req = newInnerRequest(
-        "GET",
-        this[_url],
-        () =>
-          lastEventIDValueCopy === ""
-            ? [
-              ["accept", "text/event-stream"],
-            ]
-            : [
-              ["accept", "text/event-stream"],
-              [
-                "Last-Event-Id",
-                op_utf8_to_byte_string(lastEventIDValueCopy),
-              ],
+    const req = newInnerRequest(
+      "GET",
+      this[_url],
+      () =>
+        this[_lastEventID] === ""
+          ? [
+            ["accept", "text/event-stream"],
+          ]
+          : [
+            ["accept", "text/event-stream"],
+            [
+              "Last-Event-Id",
+              op_utf8_to_byte_string(this[_lastEventID]),
             ],
-        null,
-        false,
-      );
-      /** @type {InnerResponse} */
-      const res = await mainFetch(req, true, this[_abortController].signal);
-
-      const contentType = ArrayPrototypeFind(
-        res.headerList,
-        (header) => StringPrototypeToLowerCase(header[0]) === "content-type",
-      );
-      if (res.type === "error") {
-        if (res.aborted) {
-          this[_readyState] = CLOSED;
-          this.dispatchEvent(new Event("error"));
-          break;
-        } else {
-          if (this[_readyState] === CLOSED) {
-            this[_abortController].abort();
-            break;
-          }
-          this[_readyState] = CONNECTING;
-          this.dispatchEvent(new Event("error"));
-          await new Promise((res) => setTimeout(res, this[_reconnectionTime]));
-          if (this[_readyState] !== CONNECTING) {
-            continue;
-          }
+          ],
+      null,
+      false,
+    );
+    /** @type {InnerResponse} */
+    let res;
+    try {
+      res = await mainFetch(req, true, this[_abortController].signal);
+    } catch {
+      this[_reestablishConnection]();
+      return;
+    }
 
-          if (this[_lastEventID] !== "") {
-            lastEventIDValue = this[_lastEventID];
-          }
-          continue;
-        }
-      } else if (
-        res.status !== 200 ||
-        !StringPrototypeIncludes(
-          contentType?.[1].toLowerCase(),
-          "text/event-stream",
-        )
-      ) {
-        this[_readyState] = CLOSED;
-        this.dispatchEvent(new Event("error"));
-        break;
-      }
+    if (res.aborted) {
+      this[_failConnection]();
+      return;
+    }
+    if (res.type === "error") {
+      this[_reestablishConnection]();
+      return;
+    }
+    const contentType = ArrayPrototypeFind(
+      res.headerList,
+      (header) => StringPrototypeToLowerCase(header[0]) === "content-type",
+    );
+    if (
+      res.status !== 200 ||
+      !StringPrototypeIncludes(
```

Comment:
**nitpick:** Move the ternary over the `StringPrototypeIncludes` and return a `false` from the else-branch: No reason to call the function if we know its return value.